src/applications/piezoscan.py

```diff
@@ -117,9 +124,16 @@ def set_onclick_callback(self, f):
 
     def onclick(self, event):
         if event.inaxes is self.ax:
-            #todo: draw a circle around clicked point? Maybe with a high alpha, so that its faint
             self.onclick_callback(event)
 
+            # draws a x for clicked point
+            for line in self.ax.lines:
+                if line.get_label() == 'pointer':
+                    line.set_marker('')
+            self.ax.plot(event.xdata, event.ydata,'yx', label='pointer')
```

Comment:
like above, it's probably going to be better to retain the line2D object returned by ax.plot. Then we can eliminate this for-loop. Also to note, this might be memory-leak because every time a user clicks on the screen it allocates a new chunk of memory -- it creates a new line2D object every time.  Unless Python is cleaning up for us in the background.

Reply:
ps: Following the same line of thought, I wonder if line 98 (above) is essentially a memory leak as well -- instead of calling self.ax.imshow and assigning the output to a new handle (self.artist) every time we call 'update', should we instead update the data inside of the self.artist object?  But maybe python is cleaning garbage for us. Would have to profile this.

Reply:
Done. no more for loops are used